google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -590,7 +641,8 @@ public Boolean call() {
   @Override
   public TableResult query(QueryJobConfiguration configuration, JobOption... options)
       throws InterruptedException, JobException {
-    return query(configuration, JobId.of(), options);
+    Job.checkNotDryRun(configuration, "query");
```

Comment:
Is the reason for checkNotDryRun that a TableResult is not valid / useful in this case? Is this limitation documented in the interface?

Reply:
The primary problem is that dry-run doesn't execute the query, so there's no table to return from the API. We'd essentially be forced to either return a null or artificially-empty table. I thought failing explicitly made more sense.

Clarified this in doc.